From a8baa34ace934e04c399546615038ac939884f5a Mon Sep 17 00:00:00 2001
From: Tao Huang <huangtao@rock-chips.com>
Date: Fri, 3 Apr 2020 19:06:14 +0800
Subject: [PATCH] f2fs: fix memory leak in f2fs_try_convert_inline_dir()

If fscrypt_setup_filename() successful, fscrypt_free_filename()
must be called later to clean up.

Change-Id: I761472d605a1a3c64307fc997210ca2007bd8932
Signed-off-by: Wu Liangqing <wlq@rock-chips.com>
Signed-off-by: Tao Huang <huangtao@rock-chips.com>
---
 fs/f2fs/inline.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/fs/f2fs/inline.c b/fs/f2fs/inline.c
index c68a32369f44..636f0ce1afea 100644
--- a/fs/f2fs/inline.c
+++ b/fs/f2fs/inline.c
@@ -572,7 +572,7 @@ int f2fs_try_convert_inline_dir(struct inode *dir, struct dentry *dentry)
 
 	err = fscrypt_setup_filename(dir, &dentry->d_name, 0, &fname);
 	if (err)
-		goto out;
+		goto out_unlock;
 
 	ipage = f2fs_get_node_page(sbi, dir->i_ino);
 	if (IS_ERR(ipage)) {
@@ -591,6 +591,8 @@ int f2fs_try_convert_inline_dir(struct inode *dir, struct dentry *dentry)
 	if (!err)
 		f2fs_put_page(ipage, 1);
 out:
+	fscrypt_free_filename(&fname);
+out_unlock:
 	f2fs_unlock_op(sbi);
 	return err;
 }
-- 
2.35.3

